src/common/collection.c

```diff
@@ -1331,6 +1335,133 @@ static void _dt_collection_recount_callback_2(gpointer instance, uint8_t id, gpo
   }
 }
 
+/* move images with drag and drop */
+void dt_collection_move_before(const int32_t image_id, GList * selected_images)
+{
+  if (!selected_images)
+  {
+    return;
+  }
+
+  const guint selected_images_length = g_list_length(selected_images);
+
+  if (selected_images_length == 0)
+  {
+    return;
+  }
+
+  // getting the position of the target image
+  int target_image_pos = -1;
+
+  if (image_id >= 0)
+  {
+    sqlite3_stmt *stmt = NULL;
+    gchar *image_pos_query = NULL;
+    image_pos_query = dt_util_dstrcat(
+          image_pos_query,
+          "SELECT position FROM main.images WHERE id = %i",
+          image_id);
+
+    DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), image_pos_query, -1, &stmt, NULL);
+
+    if(sqlite3_step(stmt) == SQLITE_ROW)
+    {
+      target_image_pos = sqlite3_column_int(stmt, 0);
+    }
+
+    sqlite3_finalize(stmt);
+    g_free(image_pos_query);
+  }
+
+  if (target_image_pos >= 0)
+  {
+    DT_DEBUG_SQLITE3_EXEC(dt_database_get(darktable.db), "BEGIN;", NULL, NULL, NULL);
+    sqlite3_stmt *stmt = NULL;
+
+    // shift image positions to make some space
+    gchar *update_image_pos_query = NULL;
+    update_image_pos_query = dt_util_dstrcat(
+          update_image_pos_query,
+          "UPDATE main.images SET position = position + %i WHERE position >= %i",
```

Comment:
I'm curious what the cost of this update is for someone with 100K+ images in their library (which some users will have). I've really no idea of the answer to this -- e.g. how will sqlite can handle this.

Reply:
I didn't think of that many pictures. There is an index on positions, so the update can be done at linear cost. But still, you are right, that many updates are not needed. I'll fix this.

Reply:
Here:

    > sqlite3 .config/darktable/library.db "select count(*) from images"
    55598

From reading the dt list, I'd expect that someone has a library that exceeds this by an order of magnitude.